openlibrary/plugins/openlibrary/js/index.js

```diff
@@ -443,6 +443,18 @@ jQuery(function () {
         $('details[open]').not(this).removeAttr('open');
     });
 
+    $('.header-dropdown').on('keydown', function (event) {
+        if (event.key === 'Escape') {
+            $('.header-dropdown > details[open]').not(this).removeAttr('open');
+        }
+    });
+
+    $('.dropdown-menu').each(function() {
+        $(this).find('a').last().on('focusout', function() {
+            $('.header-dropdown > details[open]').not(this).removeAttr('open');
+        });
+    });
```

Comment:
Oh I see the one's above do this 😅 We also shouldn't be using `.not(this)` here, since we actually _do_ want to close `this`!